src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,182 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    public MyLinkedList() {
+        head = tail = null;
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
+        Node<T> tempNode = new Node<>(null, value, null);
+        if (isEmpty()) {
+            addNodeToHead(tempNode);
+        } else {
+            addNodeToTail(tempNode);
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndex(index, size + 1);
+        Node<T> tempNode = new Node<>(null, value, null);
+        if (index == 0) {
+            addNodeToHead(tempNode);
+        } else {
+            Node<T> currentNode = findNodeByIndex(index - 1);
+            tempNode.next = currentNode.next;
+            tempNode.prev = currentNode;
+            currentNode.next = tempNode;
+            if (tempNode.next != null) {
+                tempNode.next.prev = tempNode;
+            }
+            if (index == size) {
```

Comment:
I beleive this logic should be outside else block, because we have 3 possible cases: addToHead, toMiddle, toTail. And this checks should be on the same level